npm_and_yarn/lib/dependabot/npm_and_yarn/version.rb

```diff
@@ -71,6 +72,18 @@ def initialize(version)
         super(T.must(version))
       end
 
+      sig { params(version: VersionParameter).returns(VersionParameter) }
+      def clean_version(version)
```

Comment:
We are cleaning version because corepack new version when installing locally, try to add packageManager field to package.json and instead of returning version it is returning warning for that then the version becomes malformed. So We are cleaning the version and finding the right version to get the version from the warning.